<launch>
    <!-- Config for panoptic mapping and the labler -->
    <arg name="config" default="3d_space_labler"/>
    <arg name="panoptic_config" default="panoptic_config"/>

    <!--  I/O arguments -->
    <arg name="experiment_name" default="PlannerWithImpactFactorCuriosity_run1"/>
    <arg name="output_folder" default="/media/rene/Empty/labeling"/>
    <arg name="experiment_path"
         default="/media/rene/Empty/presentation/Curiosity_planner_clustered_959_01__12_10_2021"/>

    <!-- Additional Params -->
    <arg name="images_to_label" default="10"/>
    <arg name="scoring_method" default="RANDOM"/> <!-- SIZE, ENTROPY, BELONGS_PROBABILITY, UNCERTAINTY-->
    <arg name="use_gt_clustering" default="true"/>
    <arg name="step" default="step_001"/>
    <arg name="rate" default="3"/> <!-- Hz -->

    <!-- ============ RUN ============== -->

    <!-- Playback -->
    <node name="dataset_player_with_gt_node" pkg="volumetric_labeling" type="dataset_replay_label_voxel_node.py"
          output="screen"
          required="true" cwd="node">

        <param name="step" value="$(arg step)"/>
        <param name="experiment_path" value="$(arg experiment_path)"/>
        <param name="replay_rate" value="$(arg rate)"/>
        <param name="out_folder" value="$(arg output_folder)"/>
        <param name="scoring_method" value="$(arg scoring_method)"/>
        <param name="images_to_label" value="$(arg images_to_label)"/>
    </node>

    <!-- Config for labler node-->
    <group ns="volumetric_labler_node">
        <group ns="panoptic">
            <rosparam file="$(find volumetric_labeling)/config/$(arg panoptic_config).yaml"/>
            <group ns="gt">
                <rosparam file="$(find volumetric_labeling)/config/$(arg panoptic_config).yaml"/>
            </group>
        </group>
    </group>

    <!-- Labler node -->
    <node name="volumetric_labler_node" type="volumetric_labler_node" pkg="volumetric_labeling" required="true"
          output="screen">
        <param name="useSpacialLabeling" value="true"/>
        <param name="scoringMethod" value="$(arg scoring_method)"/>
        <param name="use_gt_clustering" value="$(arg use_gt_clustering)"/>

        <remap from="/volumetric_labler_node/panoptic/color_image_in" to="/rgb"/>
        <remap from="/volumetric_labler_node/panoptic/depth_image_in" to="/depth"/>
        <remap from="/volumetric_labler_node/panoptic/segmentation_image_in" to="/semseg"/>
        <remap from="/volumetric_labler_node/panoptic/uncertainty_image_in" to="/uncertainty"/>

        <remap from="/volumetric_labler_node/panoptic/gt/color_image_in" to="/rgb"/>
        <remap from="/volumetric_labler_node/panoptic/gt/depth_image_in" to="/depth"/>
        <remap from="/volumetric_labler_node/panoptic/gt/segmentation_image_in" to="/gt"/>
        <remap from="/volumetric_labler_node/panoptic/gt/uncertainty_image_in" to="/uncertainty"/>
    </node>


    <node type="rviz" name="rviz" pkg="rviz" args="-d  /home/rene/Desktop/labler.rviz" output="screen"/>
</launch>
